src/main/java/VASSAL/counters/ReportState.java

```diff
@@ -136,13 +136,13 @@ public Command myKeyEvent(KeyStroke stroke) {
 
     Command c = null;
 
-    GamePiece oldPiece = (GamePiece) getProperty(Properties.SNAPSHOT);
+    java.util.Map<String, Object> oldPiece = (java.util.Map<String, Object>) getProperty(Properties.SNAPSHOT);
 
-    boolean wasVisible = oldPiece != null && !Boolean.TRUE.equals(oldPiece.getProperty(Properties.INVISIBLE_TO_OTHERS));
+    boolean wasVisible = oldPiece != null && !Boolean.TRUE.equals(oldPiece.get(Properties.INVISIBLE_TO_OTHERS));
     boolean isVisible = !Boolean.TRUE.equals(outer.getProperty(Properties.INVISIBLE_TO_OTHERS));
 
     PieceAccess.GlobalAccess.hideAll();
-    String oldUnitName = oldPiece == null ? null : oldPiece.getLocalizedName();
+    String oldUnitName = oldPiece == null ? null : (String) oldPiece.get(BasicPiece.BASIC_NAME);
```

Comment:
Not sure about this at all but it was the best replacement for `getLocalizedName()` I could find, BasicPiece also returns `getLocalizedName()` when asked for `getLocalizedProperty(BasicPiece.BASIC_NAME)`

Reply:
No, this is not going to produce the same result if a designer has provided translations for either Basic Piece Names or words in Text Label formats. getLocalizedName() is not a single property.

piece.getLocalizedName() starts at the outer and calls getLocalizedName() all the way down the trait stack and each trait can potentially add translated elements to the piece name. piece.get(BasicPiece.BASIC_NAME) just returns the raw untranslated base name from the BasicPiece trait. In reality, the only traits that modify the localizedName are Text Label and BasicPiece, but the potential is there.

The whole purpose of this exercise is to collect information for the ReportState trait, so I think the best way forward would be to create our own temporary property for the localized name.  Modify the PropertyExporter default implementation , add a new property key LOCALIZED_NAME="localizedName" and add a new property to your property map

result.put(LOCALIZED_NAME, getLocalizedName());

Then ReportState can use

String oldUnitName = oldPiece == null ? null : oldPiece.getProperty(PropertyExporter.LOCALIZED_NAME);